pkg/backend/httpstate/client/json_patch_test.go

```diff
@@ -0,0 +1,126 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	jpatch "github.com/evanphx/json-patch/v5"
+	"pgregory.net/rapid"
+)
+
+func canonicalizeJson(jsonData []byte) ([]byte, error) {
+	var m interface{}
+	if err := json.Unmarshal(jsonData, &m); err != nil {
+		return nil, err
+	}
+	canonical, err := json.Marshal(m)
+	if err != nil {
+		return nil, err
+	}
+	return canonical, nil
+}
+
+func decodePatch(raw []byte) (jpatch.Patch, error) {
+	if bytes.Equal(raw, []byte(`{}`)) {
+		return []jpatch.Operation{}, nil
+	}
+	return jpatch.DecodePatch(marshal([]json.RawMessage{raw}))
+}
+
+func TestPatchTurnaround(t *testing.T) {
```

Comment:
@justinvp you had concerns that the hashes don't match when using JSON patches. This property seems to work for me so far, though models may be a bit unhelpful, tinkering further. IN an abbreviated form, this checks this prop:

```
canonicalizeJson(patch(o, diff(o, m))) == canonicalizeJson(m)
```

Reply:
Thanks for exploring this! It's totally possible that I was doing something silly in the prototype (i.e. not canonicalizing), or I was running into a bug in that library last year.